### ISSUE TITLE
reboot if needed, even if patching fails

### ISSUE THREAD

**Issue by jcpunk** [puppetlabs/puppetlabs-patching_as_code#63]
## Use Case
When a linux system has a pending reboot, but patching fails, the system is not rebooted.  This can happen if a 3rd party repo isn't fully up to date.  Some patches can get installed that would benefit from a reboot, but, because the patching failed, the system is not checked for a pending reboot.

## Describe the Solution You Would Like
Perhaps make the reboot logic smarter to so reboot ifneeded can trigger under this workflow?

## Describe Alternatives You've Considered
I don't have a great alternate workflow I can think of...

## Additional Context
This might need an additional flag to toggle the behavior, I'm not sure...

**Comment by kreeuwijk** [comment]
I don't think this is easily possible from within the module, since we're using the native `package` resource to update packages. As soon as one fails, Puppet treats it as a run failure and will refuse to apply subsequent dependent resources & stages.

You'd probably have to look at a custom report processor on the Puppet server to analyze the incoming agent run report and determine a patch failure. You could theoretically then make that custom report processor launch a PE/Bolt Task to restart the system. This is outside of the scope of this module though.

**Comment by jcpunk** [comment]
Sounds reasonable to moe
